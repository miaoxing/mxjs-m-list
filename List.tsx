import * as React from 'react';
import clsx from 'clsx';
import {View} from '@fower/taro';

export interface ListProps {
  as?: React.ElementType;
  className?: string;

  /**
   * 紧凑列表
   */
  size?: '' | 'sm';

  /**
   * 无边框列表
   */
  borderless?: boolean;
}

const List = (props: ListProps): JSX.Element => {
  const {
    as: Component,
    className,
    size,
    borderless,
    ...rest
  } = props;

  return (
    <Component
      {...rest}
      className={clsx(
        className,
        'mx-list',
        size === 'sm' && 'mx-list-sm',
        borderless && 'mx-list-borderless',
      )}
    />
  );
};

List.defaultProps = {
  as: View,
};

export default List;
