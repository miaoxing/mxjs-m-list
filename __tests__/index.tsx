import * as React from 'react';
import {render} from '@testing-library/react';
import List from '../List';

describe('list', () => {
  test('List', () => {
    const result = render(<List/>);
    expect(result.container).toMatchSnapshot();
  });
});
