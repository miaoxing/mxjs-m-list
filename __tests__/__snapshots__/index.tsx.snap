// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list List 1`] = `
<div>
  <taro-view-core
    class="mx-list"
  />
</div>
`;
